src/modes/ModeEstimation.jl

```diff
@@ -0,0 +1,235 @@
+module ModeEstimation
```

Comment:
I guess this is motivated by the fact that it might become a separate package at some point? IMO having nested modules is usually not very convenient for handling imports and exports (and debugging...), so I try to avoid it.

Reply:
Well, we only export two things, and I'm hoping this is not going to live in Turing for very long, so I wanted to keep all of these things separate.